website/blog/how-to-migrate-an-ios-app-to-bazel.md

```diff
@@ -0,0 +1,1470 @@
+[spawn_strategy-flag]: https://bazel.build/versions/7.0.0/reference/command-line-reference#flag--spawn_strategy
+
+## Optimizing
+
+Now that everything is building,
+and we have a functioning remote cache,
+it's time to optimize the build further.
+Ideally we wouldn't need to do anything further at this point,
+but currently Bazel doesn't have the best default settings for all projects.
+And even if it did,
+the structure of the build graph itself has a large impact on build performance.
+
+### Optimal Bazel settings
+
+If you build with Xcode,
+**rules_xcodeproj** will apply a lot of [optimal settings for you][rules_xcodeproj-default-flags].
+I'll still list them here,
+along with some settings it doesn't set for you,
+in case you need to build outside of Xcode.
+
+At [this commit][commit-optimal-flags] we applied all of the flags discussed in the following sections to the Mastodon iOS project.
+
+[commit-optimal-flags]: https://github.com/brentleyjones/mastodon-ios/commit/3b2b3f6f1147b8c0369f0924b6869ad08de1cdd3
+[rules_xcodeproj-default-flags]: https://github.com/MobileNativeFoundation/rules_xcodeproj/blob/1.17.0/xcodeproj/internal/templates/xcodeproj.bazelrc
+
+#### Cacheability
+
+The
+[`oso_prefix_is_pwd`][oso_prefix_is_pwd-feature],
+[`relative_ast_path`][relative_ast_path-feature],
+and [`remap_xcode_path`][remap_xcode_path-feature]
+[features][features-flag] in **apple_support**,
+and the [`swift.cacheable_swiftmodules`][swift-cacheable_swiftmodules-feature] feature in **rules_swift**,
+remove absolute paths from binaries produced by **rules_apple** and **rules_swift**.
+It's highly recommended that you use these features if using a remote cache,
+otherwise you might get a low cache hit rate on anything involving those binaries.
+As long as you are using a version of **apple_support** and **rules_swift** that is at least 1.5.0,
+these features are enabled by default.
+Otherwise,
+you can enable them with `--features=oso_prefix_is_pwd,relative_ast_path,remap_xcode_path,swift.cacheable_swiftmodules`.
+
+:::note
+
+When using these features you may need to [provide additional information to lldb][swift-debug-settings] for debugging to work properly.
+
+[swift-debug-settings]: https://github.com/MobileNativeFoundation/rules_xcodeproj/blob/1.17.0/tools/generators/swift_debug_settings/src/Generator/WriteSwiftDebugSettings.swift#L121-L155
+
+:::
+
+[features-flag]: https://bazel.build/versions/7.0.0/reference/command-line-reference#flag--features
+[oso_prefix_is_pwd-feature]: https://github.com/bazelbuild/apple_support/blob/1.14.0/crosstool/cc_toolchain_config.bzl#L2080-L2089
+[relative_ast_path-feature]: https://github.com/bazelbuild/apple_support/blob/1.14.0/crosstool/cc_toolchain_config.bzl#L1747-L1761
+[remap_xcode_path-feature]: https://github.com/bazelbuild/apple_support/blob/1.14.0/crosstool/cc_toolchain_config.bzl#L1851-L1873
```

Comment:
We should make the documentation of these types of flags easier to access, IMO. Work for the future!

Reply:
Agreed.